pandas-stubs/_typing.pyi

```diff
@@ -310,25 +310,26 @@ UsecolsArgType: TypeAlias = (
     | None
 )
 # Scratch types for generics
+
 S1 = TypeVar(
     "S1",
-    str,
-    bytes,
-    datetime.date,
-    datetime.time,
-    bool,
-    int,
-    float,
-    complex,
-    Dtype,
-    Timestamp,
-    Timedelta,
-    Period,
-    Interval[int],
-    Interval[float],
-    Interval[Timestamp],
-    Interval[Timedelta],
-    CategoricalDtype,
+    bound=str
```

Comment:
See point #3 at https://github.com/microsoft/pyright/issues/5178#issuecomment-1566190533 ,  Recommended to use a `Union` rather than a constrained type var.

Reply:
Does this also affect other TypeVars (could be done later)?

Reply:
Can we use `S1 = TypeVar("S1", bound=Scalar)`? You replaced `S1` with `Scalar` in some places

Reply:
```py
 S1 = TypeVar(
     "S1",
    bound=Scalar
     | Interval[int]
     | Interval[float]
     | Interval[Timestamp]
     | Interval[Timedelta]
    | Dtype
    | Period
    | CategoricalDtype
    | datetime.time,
 )
```

Seems to work (but Scalar includes a few more types!)

Reply:
> Does this also affect other TypeVars (could be done later)?

Yes, and I created an issue for that #719